spec/util.spec.js

```diff
@@ -0,0 +1,131 @@
+import { util } from '../src/util/util';
+
+const now = new Date().toISOString();
+const playhead = '00:12:30.212';
+const assetURI = 'http://example.com/linear-asset.mp4?foo=1&bar=_-[{bar';
+
+const encodeRFC3986 = str => util.encodeURIComponentRFC3986(str);
+
+const encodedAssetURI = encodeRFC3986(assetURI);
+const encodedPlayhead = encodeRFC3986(playhead);
+const encodedTimestamp = encodeRFC3986(now);
+
+const resolve = (URLTemplate, variables) =>
+  util.resolveURLTemplates([URLTemplate], variables)[0];
+
+describe('util', function() {
```

Comment:
maybe we can avoid having this first describe? since we don't have a root `beforeEach` it might be nice to remove a level of indentation to all this code.

But then of course you'll not have these tests grouped under a `util` section, up to you

Reply:
I will leave it because I think it is clearer if it is under utils, besides jest has a nice command line interface, so it won't look messy